Software/Backend/WebApi/Controllers/JobController.cs

```diff
@@ -0,0 +1,42 @@
+using BusinessLogicLayer.AppLogic.Jobs;
+using BusinessLogicLayer.AppLogic.Jobs.AddJob;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+
+namespace WebApi.Controllers;
+
+[ApiController]
+[Route("jobs")]
+public class JobController : ControllerBase
+{
+    private readonly IJobService _jobService;
+    public JobController(IJobService jobService)
+    {
+        _jobService = jobService;
+    }
+
+    // POST: /jobs/add
+    [HttpPost("add")]
+    [Authorize]
+    public ActionResult<AddJobResponse> AddJob([FromBody] AddJobRequest request)
+    {
+        var userIdFromJwt = HttpContext.Items["UserId"] as int?;
+
+        if (userIdFromJwt == null)
+        {
+            return Unauthorized(new AddJobResponse()
+            {
+                Error = "Ne mo�ete pristupiti tom resursu!"
+            });
+        }
+
+        var job = _jobService.AddJob(request, userIdFromJwt.Value);
+
+        if (job.Error != null)
```

Comment:
Probaj umjesto (job.Error != null) staviti (!string.IsNullOrEmpty(job.Error)), jer zna biti da stavi da je error "", što NIJE null (a nema grešaka, posao se unese), pa se vrti ovo BadRequest(job).